code/controllers/subsystem/air.dm

```diff
@@ -1,35 +1,436 @@
-	wait          = 2 SECONDS
-	flags         = SS_NO_TICK_CHECK
+	wait          = 1 SECONDS
 	display_order = SS_DISPLAY_AIR
 
+	var/list/currentrun
+	var/currentpart = SSAIR_TILES
+
+	var/list/cost_parts = list(SSAIR_TILES     = 0,\
+	                           SSAIR_DEFERRED  = 0,\
+	                           SSAIR_EDGES     = 0,\
+/*	                           SSAIR_FIRE_ZONE = 0,*/\
+	                           SSAIR_HOTSPOT   = 0,\
+	                           SSAIR_ZONE      = 0)
+
+	var/list/zones = list()
+	var/list/edges = list()
+
+	//Geometry process lists
+	var/list/processing_parts = list(SSAIR_TILES     = list(),\
+	                                 SSAIR_DEFERRED  = list(),\
+	                                 SSAIR_EDGES     = list(),\
+/*	                                 SSAIR_FIRE_ZONE = list(),*/\
+	                                 SSAIR_HOTSPOT   = list(),\
+	                                 SSAIR_ZONE      = list())
+
+	var/active_zones = 0
+
+	var/current_cycle = 0
+	var/update_delay = 5 //How long between check should it try to process atmos again.
+	var/failed_ticks = 0 //How many ticks have runtimed?
+
+	var/next_id = 1 //Used to keep track of zone UIDs.
+
+
+
 /datum/subsystem/air/New()
 	NEW_SS_GLOBAL(SSair)
 
 
-/datum/subsystem/air/stat_entry()
-	if (air_master)
-		..("Z:[air_master.zones.len]|ZU:[air_master.zones_to_update.len]|ZA:[air_master.active_zones]|TU:[air_master.tiles_to_update.len]|E:[air_master.edges.len]|F:[air_master.active_hotspots.len]")
-	else
-		..("AIR MASTER DOES NOT EXIST.")
+/datum/subsystem/air/stat_entry(msg)
+	var/list/p_tiles = processing_parts[SSAIR_TILES]
+	var/list/p_zone = processing_parts[SSAIR_ZONE]
+//	var/list/p_fire_zone = processing_parts[SSAIR_FIRE_ZONE]
+	var/list/p_hotspot = processing_parts[SSAIR_HOTSPOT]
+	var/list/p_edges = processing_parts[SSAIR_EDGES]
+
+	msg += "C:{\
+	        T:[round(cost_parts[SSAIR_TILES], 1)]|\
+	        D:[round(cost_parts[SSAIR_DEFERRED], 1)]|\
+	        E:[round(cost_parts[SSAIR_EDGES], 1)]|\
+"/*	        F:[round(cost_parts[SSAIR_FIRE_ZONE], 1)]|*/+"\
+	        H:[round(cost_parts[SSAIR_HOTSPOT], 1)]|\
+	        Z:[round(cost_parts[SSAIR_ZONE], 1)]|\
+	        } T:{\
+	        Z:[zones.len]|\
+	        E:[edges.len]\
+	        } \
+	        T:[p_tiles.len]|\
+	        Z:[p_zone.len]|\
+"/*	        F:[p_fire_zone.len]|*/+"\
+	        H:[p_hotspot.len]|\
+	        E:[p_edges.len]|\
+	        A:[active_zones]"
+	..(msg) //Note to self: Don't fuck that up when uncommenting after adding fire zones
+
 
 /datum/subsystem/air/Initialize(timeofday)
-	if (!air_master)
-		air_master = new
-		air_master.Setup()
+	to_chat(world, "<span class='danger'>Processing Geometry...</span>")
+	sleep(-1)
+
+	var/simulated_turf_count = 0
+
+	for(var/turf/simulated/S in world)
+		simulated_turf_count++
+		S.update_air_properties()
+
+	processing_parts[SSAIR_EDGES] = edges //A temporary hack to make edges actually work before the later PR to add edge sleeping, etc.
```

Comment:
This is the particularly scary hack, by the way. It's scary because it relies on `processing_parts[SSAIR_EDGES]` remaining an alias of `edges`. Nothing currently resets either of those, though, so it should be fine.